src/sqlancer/ASTBasedReducer.java

```diff
@@ -0,0 +1,470 @@
+                                e.printStackTrace();
+                            }
+                        });
+                    } else if (astNode instanceof GroupByElement) {
+                        GroupByElement groupByElement = (GroupByElement) astNode;
+                        ExpressionList expressionList = groupByElement.getGroupByExpressionList();
+                        if (expressionList == null) {
+                            groupByElement.getGroupingSets();
+                            // TODO: TO BE IMPLEMENTED.
+                        } else {
+                            List<Expression> elms = expressionList.getExpressions();
+                            if (elms.size() <= 1) {
+                                continue;
+                            }
+                            listElementRemovingReduce(groupByElement, elms, (select, items) -> {
+                                groupByElement.setGroupByExpressionList(new ExpressionList(items));
+                                updateStatements();
+                            });
+                        }
+                    }
+                } catch (Exception ignoredException) {
+                }
+            }
+
+            for (String astNodeName : descendList) {
+                try {
+                    Method nodeGetter = plainSelect.getClass().getMethod(getterName(astNodeName));
+                    Object astNode = nodeGetter.invoke(plainSelect);
+                    if (astNode == null) {
+                        continue;
+                    }
+                    if (astNode instanceof Expression) {
+                        ((Expression) astNode).accept(expressionReducerVisitor);
+                    } else if (astNode instanceof List) {
+                        // Really hacky... Some other ways to simplify it ?
+                        List<?> elms = (List<?>) astNode;
+                        for (Object obj : elms) {
+                            if (obj instanceof SelectExpressionItem) {
+                                ((SelectExpressionItem) obj).accept(expressionReducerVisitor);
+                            } else if (obj instanceof Expression) {
+                                ((Expression) obj).accept(expressionReducerVisitor);
+                            }
+                        }
+                    } else if (astNode instanceof GroupByElement) {
+                        GroupByElement groupByElement = (GroupByElement) astNode;
+                        ExpressionList expressionList = groupByElement.getGroupByExpressionList();
+                        if (expressionList != null) {
+                            expressionList.accept(expressionReducerVisitor);
+                        } else {
+                            // TODO: TO BE IMPLEMENTED
+                        }
+                    }
+                } catch (Exception e) {
+                    System.out.println("Reflection Exception:");
```

Comment:
perhaps we can just throw an `AssertionError` here instead, passing `e` to its constructor.